### PR TITLE
feat(perp-liquidatable) Integrate funding rate applier

### LINKED ISSUE #2024: Build Liquidatable
**Summary**

This contract will enable a position managers' sponsors to be liquidated and for those liquidations to be disputed.

Unlike the expiring contract, the perpetual contract should incorporate a funding rate multiplier in its token redemption value (TRV) calculations. Note that a TRV can only be derived after a DVM vote resolution since it relies on an external price.

New production contracts to write:
- `Liquidatable`

**Implementation**

Proposed construction parameters must also be set up:
```
constructionParams: {
       ...All params from PositionManager,
        uint256 liquidationLiveness, // Pending liquidation time before expiry
        FixedPoint.Unsigned collateralRequirement, // Required collateral:TRV ratio for a sufficiently collateralized position
        FixedPoint.Unsigned disputeBondPct, // % of locked collateral to be deposited by a disputer
        FixedPoint.Unsigned sponsorDisputeRewardPct, // % of TRV paid to sponsor following successful dispute
        FixedPoint.Unsigned disputerDisputeRewardPc, // % of TRV paid to disputer following successful dispute
}
```

**Differences from EMP**
- TRV calculation in `_settle` should incorporate the `FundingRateMultiplier`'s current multiplier.
- no expiry logic
- Funding rate multiplier's current rate should be updated on every contract call.

***Apply Funding Rates***
- Every contract call should update `FundingRateApplier`'s rate, similar to how a regular fee is updated one very contract interaction.

**How**
Most of this can be borrowed from the EMP's PPM. New logic is fetching the funding rate and applying it to modify sponsor debt.

**Resolution**

Unit tests should be made using hard coded funding rates.

**Difficulty Score [1-10]**

- 8: ~1 week

### LINKED ISSUE #2047: Liquidatable: apply funding rate multiplier to TRV or price.
**Why**

Project Evergreen.

**Difficulty Score [1-10]**

- 6: ~1 day

### PR DESCRIPTION
**Pull request by nicholaspai**
Signed-off-by: Nick Pai <npai.nyc@gmail.com>

**Motivation**

#2024
#2047 

**Summary**

All liquidatable public methods should update funding rate. 

Funding rate multiplier should modify amount of token debt counted for token redemption values on dispute settlements.

**Issue(s)**

Close #2024
Close #2047

### CODE REVIEW

**Comment by nicholaspai** [comment]
Running into a bytecode error right now:

```sh
➜  core git:(npai/liquidatable-funding-rate) yarn truffle exec ./scripts/local/CalculateContractBytecode.js --network mainnet_mnemonic --contract PerpetualLib
yarn run v1.22.5
$ /Users/nicholaspai/UMA/protocol/node_modules/.bin/truffle exec ./scripts/local/CalculateContractBytecode.js --network mainnet_mnemonic --contract PerpetualLib
> Warning: possible unsupported (undocumented in help) command line option: --contract
Using network 'mainnet_mnemonic'.

loading PerpetualLib.json
Contract is 25664 bytes in size.
This leaves a total of -1088 bytes within the EIP170 limit.
✨  Done in 3.04s.
```

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualLiquidatable.sol [hunk 1]
```diff
@@ -417,9 +420,9 @@ contract PerpetualLiquidatable is PerpetualPositionManager {
         // Note: all payouts are scaled by the unit collateral value so all payouts are charged the fees pro rata.
         FixedPoint.Unsigned memory feeAttenuation = _getFeeAdjustedCollateral(liquidation.rawUnitCollateral);
         FixedPoint.Unsigned memory settlementPrice = liquidation.settlementPrice;
-        FixedPoint.Unsigned memory tokenRedemptionValue = liquidation.tokensOutstanding.mul(settlementPrice).mul(
-            feeAttenuation
-        );
+        FixedPoint.Unsigned memory tokenRedemptionValue = _getFundingRateAppliedTokenDebt(liquidation.tokensOutstanding)
+            .mul(settlementPrice)
```
Comment (daywiss) [inline packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualLiquidatable.sol:424]: where is _getFundingRateAppliedTokenDebt defined

Comment (nicholaspai) [reply packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualLiquidatable.sol:424]: In `FundingRateApplier`. It returns `cumulativeFeeMultiplier * tokensOutstanding`

**Pull request merged**